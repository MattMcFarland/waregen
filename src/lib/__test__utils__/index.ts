import { clone, range } from "@@/utils";
import { DefaultWareEntity, DefaultBlueprintEntity } from "@@/entities/Ware";
import { AddwareEntity } from "@@/XMLTypes/X4WareGenXML";
import { GeneratorConfig } from "@@/generator";

export const mockGeneratorConfig = (): GeneratorConfig => ({
  defaultWare: defaultWare(),
  defaultBlueprint: defaultBlueprint(),
  addwaresList: mockAddwares(4),
  gamePath: "/test/Game",
  modPath: "extensions/test",
  modPrefix: "pfx",
  unpackedPath: "unpacked"
});

export const mockAddwares = (count: number): AddwareEntity[] => {
  const char = (code: number) => "abcdefghijklmnopqrstuvwxyz"[code];

  return range(count).map(
    (i: number): AddwareEntity => ({
      Attributes: {
        baskets: `[test_basket_${char(i)}]`,
        cloneProductionModuleFrom: `prod_cloned_macro_${char(i)}`,
        id: `${char(i)}`
      },
      ware: [mockWare(`${char(i)}`)],
      blueprint: [mockBlueprint(`${char(i)}`)]
    })
  );
};

export const mockWare = (name: string) => {
  const ware = defaultWare();
  ware.Attributes.name = name;
  return clone(ware);
};
export const mockBlueprint = (name: string) => {
  const bp = defaultBlueprint();
  bp.Attributes.name = `${name} production`;
  return clone(bp);
};

export const defaultBlueprint = (): DefaultBlueprintEntity => {
  const bp = <DefaultBlueprintEntity>defaultWare();
  bp.Attributes.tags = "module";
  bp.Attributes.description = "blueprint description";
  bp.restriction = [
    {
      Attributes: { licence: "station_gen_basic " }
    }
  ];
  bp.research = [
    {
      Attributes: { time: 10 },
      research: [
        { ware: [{ Attributes: { ware: "research_module_production" } }] }
      ]
    }
  ];
  bp.owner = [
    { Attributes: { faction: "antigone" } },
    { Attributes: { faction: "argon" } },
    { Attributes: { faction: "holyorder" } },
    { Attributes: { faction: "paranid" } },
    { Attributes: { faction: "teladi" } }
  ];
  bp.production = [
    {
      Attributes: {
        time: 791,
        amount: 1,
        method: "default",
        name: "{20206,101}"
      },
      primary: [
        {
          ware: [
            { Attributes: { ware: "claytronics", amount: 30 } },
            { Attributes: { ware: "energycells", amount: 75 } },
            { Attributes: { ware: "hullparts", amount: 150 } }
          ]
        }
      ]
    }
  ];
  return clone(bp);
};

export const defaultWare = (): DefaultWareEntity =>
  clone({
    Attributes: {
      id: "default",
      name: "default",
      transport: "container",
      volume: 1
    },
    price: [
      {
        Attributes: {
          min: 1,
          average: 1,
          max: 1
        }
      }
    ],
    production: [
      {
        Attributes: {
          time: 10,
          amount: 1,
          method: "default",
          name: "{20206, 101}"
        },
        primary: [
          {
            ware: [{ Attributes: { ware: "energycells", amount: 50 } }]
          }
        ],
        effects: [
          {
            effect: [
              {
                Attributes: { type: "efficiency", product: 1 }
              }
            ]
          }
        ]
      }
    ]
  });
