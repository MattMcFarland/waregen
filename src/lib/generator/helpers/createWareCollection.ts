import { WareCollection, Ware } from "../../entities";
import { GeneratorConfig } from "../../generator";
import { IdRoster } from "../../utils/IdRoster";
import idx from "idx";
import { WareOrBlueprintEntity } from "../../entities/Ware";

export default function createWares(config: GeneratorConfig): WareCollection {
  const wares = config.addwaresList.map(addware => {
    const ids = new IdRoster(addware.Attributes.id, config.modPrefix);
    const wareDef = <WareOrBlueprintEntity>idx(addware, _ => _.ware[0]);
    const ware = new Ware(wareDef, config.defaultWare);
    ware.id = ids.ware;
    return ware;
  });
  const blueprintWares = config.addwaresList.map(addware => {
    const ids = new IdRoster(addware.Attributes.id, config.modPrefix);
    const wareDef = <WareOrBlueprintEntity>idx(addware, _ => _.blueprint[0]);
    const bpWare = new Ware(wareDef, config.defaultBlueprint);
    bpWare.id = ids.productionBlueprintId;
    bpWare.componentRef = ids.productionMacro;
    bpWare.componentAmount = 1;
    return bpWare;
  });
  return new WareCollection([...wares, ...blueprintWares]);
}
