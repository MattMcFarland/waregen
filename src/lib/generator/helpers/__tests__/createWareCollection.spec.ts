import { mockGeneratorConfig } from "@@/__test__utils__";
import createWareCollection from "../createWareCollection";

describe("generator/helpers", () => {
  describe("createWareCollection", () => {
    test("toJson()", () => {
      const mockConfig = mockGeneratorConfig();
      const wareCollection = createWareCollection(mockConfig);
      expect(wareCollection.toJson()).toMatchSnapshot();
    });
    test("toXml()", () => {
      const mockConfig = mockGeneratorConfig();
      const wareCollection = createWareCollection(mockConfig);
      expect(wareCollection.toXml()).toMatchSnapshot();
    });
  });
});
