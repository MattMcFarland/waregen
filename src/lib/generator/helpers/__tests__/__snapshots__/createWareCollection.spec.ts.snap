// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator/helpers createWareCollection toJson() 1`] = `"{\\"ware\\":[{\\"Attributes\\":{\\"id\\":\\"pfx_a\\",\\"name\\":\\"a\\",\\"transport\\":\\"container\\",\\"volume\\":1},\\"price\\":[{\\"Attributes\\":{\\"min\\":1,\\"average\\":1,\\"max\\":1}}],\\"production\\":[{\\"Attributes\\":{\\"time\\":10,\\"amount\\":1,\\"method\\":\\"default\\",\\"name\\":\\"{20206, 101}\\"},\\"primary\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"energycells\\",\\"amount\\":50}}]}],\\"effects\\":[{\\"effect\\":[{\\"Attributes\\":{\\"type\\":\\"efficiency\\",\\"product\\":1}}]}]}]},{\\"Attributes\\":{\\"id\\":\\"pfx_b\\",\\"name\\":\\"b\\",\\"transport\\":\\"container\\",\\"volume\\":1},\\"price\\":[{\\"Attributes\\":{\\"min\\":1,\\"average\\":1,\\"max\\":1}}],\\"production\\":[{\\"Attributes\\":{\\"time\\":10,\\"amount\\":1,\\"method\\":\\"default\\",\\"name\\":\\"{20206, 101}\\"},\\"primary\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"energycells\\",\\"amount\\":50}}]}],\\"effects\\":[{\\"effect\\":[{\\"Attributes\\":{\\"type\\":\\"efficiency\\",\\"product\\":1}}]}]}]},{\\"Attributes\\":{\\"id\\":\\"pfx_c\\",\\"name\\":\\"c\\",\\"transport\\":\\"container\\",\\"volume\\":1},\\"price\\":[{\\"Attributes\\":{\\"min\\":1,\\"average\\":1,\\"max\\":1}}],\\"production\\":[{\\"Attributes\\":{\\"time\\":10,\\"amount\\":1,\\"method\\":\\"default\\",\\"name\\":\\"{20206, 101}\\"},\\"primary\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"energycells\\",\\"amount\\":50}}]}],\\"effects\\":[{\\"effect\\":[{\\"Attributes\\":{\\"type\\":\\"efficiency\\",\\"product\\":1}}]}]}]},{\\"Attributes\\":{\\"id\\":\\"pfx_d\\",\\"name\\":\\"d\\",\\"transport\\":\\"container\\",\\"volume\\":1},\\"price\\":[{\\"Attributes\\":{\\"min\\":1,\\"average\\":1,\\"max\\":1}}],\\"production\\":[{\\"Attributes\\":{\\"time\\":10,\\"amount\\":1,\\"method\\":\\"default\\",\\"name\\":\\"{20206, 101}\\"},\\"primary\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"energycells\\",\\"amount\\":50}}]}],\\"effects\\":[{\\"effect\\":[{\\"Attributes\\":{\\"type\\":\\"efficiency\\",\\"product\\":1}}]}]}]},{\\"Attributes\\":{\\"id\\":\\"pfx_e\\",\\"name\\":\\"e\\",\\"transport\\":\\"container\\",\\"volume\\":1},\\"price\\":[{\\"Attributes\\":{\\"min\\":1,\\"average\\":1,\\"max\\":1}}],\\"production\\":[{\\"Attributes\\":{\\"time\\":10,\\"amount\\":1,\\"method\\":\\"default\\",\\"name\\":\\"{20206, 101}\\"},\\"primary\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"energycells\\",\\"amount\\":50}}]}],\\"effects\\":[{\\"effect\\":[{\\"Attributes\\":{\\"type\\":\\"efficiency\\",\\"product\\":1}}]}]}]},{\\"Attributes\\":{\\"id\\":\\"pfx_prod_gen_a_macro_01\\",\\"name\\":\\"a production\\",\\"transport\\":\\"container\\",\\"volume\\":1,\\"tags\\":\\"module\\",\\"description\\":\\"blueprint description\\"},\\"price\\":[{\\"Attributes\\":{\\"min\\":1,\\"average\\":1,\\"max\\":1}}],\\"production\\":[{\\"Attributes\\":{\\"time\\":791,\\"amount\\":1,\\"method\\":\\"default\\",\\"name\\":\\"{20206,101}\\"},\\"primary\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"claytronics\\",\\"amount\\":30}},{\\"Attributes\\":{\\"ware\\":\\"energycells\\",\\"amount\\":75}},{\\"Attributes\\":{\\"ware\\":\\"hullparts\\",\\"amount\\":150}}]}]}],\\"restriction\\":[{\\"Attributes\\":{\\"licence\\":\\"station_gen_basic \\"}}],\\"research\\":[{\\"Attributes\\":{\\"time\\":10},\\"research\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"research_module_production\\"}}]}]}],\\"owner\\":[{\\"Attributes\\":{\\"faction\\":\\"antigone\\"}},{\\"Attributes\\":{\\"faction\\":\\"argon\\"}},{\\"Attributes\\":{\\"faction\\":\\"holyorder\\"}},{\\"Attributes\\":{\\"faction\\":\\"paranid\\"}},{\\"Attributes\\":{\\"faction\\":\\"teladi\\"}}],\\"component\\":[{\\"Attributes\\":{\\"amount\\":1,\\"ref\\":\\"pfx_prod_gen_a_macro\\"}}]},{\\"Attributes\\":{\\"id\\":\\"pfx_prod_gen_b_macro_01\\",\\"name\\":\\"b production\\",\\"transport\\":\\"container\\",\\"volume\\":1,\\"tags\\":\\"module\\",\\"description\\":\\"blueprint description\\"},\\"price\\":[{\\"Attributes\\":{\\"min\\":1,\\"average\\":1,\\"max\\":1}}],\\"production\\":[{\\"Attributes\\":{\\"time\\":791,\\"amount\\":1,\\"method\\":\\"default\\",\\"name\\":\\"{20206,101}\\"},\\"primary\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"claytronics\\",\\"amount\\":30}},{\\"Attributes\\":{\\"ware\\":\\"energycells\\",\\"amount\\":75}},{\\"Attributes\\":{\\"ware\\":\\"hullparts\\",\\"amount\\":150}}]}]}],\\"restriction\\":[{\\"Attributes\\":{\\"licence\\":\\"station_gen_basic \\"}}],\\"research\\":[{\\"Attributes\\":{\\"time\\":10},\\"research\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"research_module_production\\"}}]}]}],\\"owner\\":[{\\"Attributes\\":{\\"faction\\":\\"antigone\\"}},{\\"Attributes\\":{\\"faction\\":\\"argon\\"}},{\\"Attributes\\":{\\"faction\\":\\"holyorder\\"}},{\\"Attributes\\":{\\"faction\\":\\"paranid\\"}},{\\"Attributes\\":{\\"faction\\":\\"teladi\\"}}],\\"component\\":[{\\"Attributes\\":{\\"amount\\":1,\\"ref\\":\\"pfx_prod_gen_b_macro\\"}}]},{\\"Attributes\\":{\\"id\\":\\"pfx_prod_gen_c_macro_01\\",\\"name\\":\\"c production\\",\\"transport\\":\\"container\\",\\"volume\\":1,\\"tags\\":\\"module\\",\\"description\\":\\"blueprint description\\"},\\"price\\":[{\\"Attributes\\":{\\"min\\":1,\\"average\\":1,\\"max\\":1}}],\\"production\\":[{\\"Attributes\\":{\\"time\\":791,\\"amount\\":1,\\"method\\":\\"default\\",\\"name\\":\\"{20206,101}\\"},\\"primary\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"claytronics\\",\\"amount\\":30}},{\\"Attributes\\":{\\"ware\\":\\"energycells\\",\\"amount\\":75}},{\\"Attributes\\":{\\"ware\\":\\"hullparts\\",\\"amount\\":150}}]}]}],\\"restriction\\":[{\\"Attributes\\":{\\"licence\\":\\"station_gen_basic \\"}}],\\"research\\":[{\\"Attributes\\":{\\"time\\":10},\\"research\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"research_module_production\\"}}]}]}],\\"owner\\":[{\\"Attributes\\":{\\"faction\\":\\"antigone\\"}},{\\"Attributes\\":{\\"faction\\":\\"argon\\"}},{\\"Attributes\\":{\\"faction\\":\\"holyorder\\"}},{\\"Attributes\\":{\\"faction\\":\\"paranid\\"}},{\\"Attributes\\":{\\"faction\\":\\"teladi\\"}}],\\"component\\":[{\\"Attributes\\":{\\"amount\\":1,\\"ref\\":\\"pfx_prod_gen_c_macro\\"}}]},{\\"Attributes\\":{\\"id\\":\\"pfx_prod_gen_d_macro_01\\",\\"name\\":\\"d production\\",\\"transport\\":\\"container\\",\\"volume\\":1,\\"tags\\":\\"module\\",\\"description\\":\\"blueprint description\\"},\\"price\\":[{\\"Attributes\\":{\\"min\\":1,\\"average\\":1,\\"max\\":1}}],\\"production\\":[{\\"Attributes\\":{\\"time\\":791,\\"amount\\":1,\\"method\\":\\"default\\",\\"name\\":\\"{20206,101}\\"},\\"primary\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"claytronics\\",\\"amount\\":30}},{\\"Attributes\\":{\\"ware\\":\\"energycells\\",\\"amount\\":75}},{\\"Attributes\\":{\\"ware\\":\\"hullparts\\",\\"amount\\":150}}]}]}],\\"restriction\\":[{\\"Attributes\\":{\\"licence\\":\\"station_gen_basic \\"}}],\\"research\\":[{\\"Attributes\\":{\\"time\\":10},\\"research\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"research_module_production\\"}}]}]}],\\"owner\\":[{\\"Attributes\\":{\\"faction\\":\\"antigone\\"}},{\\"Attributes\\":{\\"faction\\":\\"argon\\"}},{\\"Attributes\\":{\\"faction\\":\\"holyorder\\"}},{\\"Attributes\\":{\\"faction\\":\\"paranid\\"}},{\\"Attributes\\":{\\"faction\\":\\"teladi\\"}}],\\"component\\":[{\\"Attributes\\":{\\"amount\\":1,\\"ref\\":\\"pfx_prod_gen_d_macro\\"}}]},{\\"Attributes\\":{\\"id\\":\\"pfx_prod_gen_e_macro_01\\",\\"name\\":\\"e production\\",\\"transport\\":\\"container\\",\\"volume\\":1,\\"tags\\":\\"module\\",\\"description\\":\\"blueprint description\\"},\\"price\\":[{\\"Attributes\\":{\\"min\\":1,\\"average\\":1,\\"max\\":1}}],\\"production\\":[{\\"Attributes\\":{\\"time\\":791,\\"amount\\":1,\\"method\\":\\"default\\",\\"name\\":\\"{20206,101}\\"},\\"primary\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"claytronics\\",\\"amount\\":30}},{\\"Attributes\\":{\\"ware\\":\\"energycells\\",\\"amount\\":75}},{\\"Attributes\\":{\\"ware\\":\\"hullparts\\",\\"amount\\":150}}]}]}],\\"restriction\\":[{\\"Attributes\\":{\\"licence\\":\\"station_gen_basic \\"}}],\\"research\\":[{\\"Attributes\\":{\\"time\\":10},\\"research\\":[{\\"ware\\":[{\\"Attributes\\":{\\"ware\\":\\"research_module_production\\"}}]}]}],\\"owner\\":[{\\"Attributes\\":{\\"faction\\":\\"antigone\\"}},{\\"Attributes\\":{\\"faction\\":\\"argon\\"}},{\\"Attributes\\":{\\"faction\\":\\"holyorder\\"}},{\\"Attributes\\":{\\"faction\\":\\"paranid\\"}},{\\"Attributes\\":{\\"faction\\":\\"teladi\\"}}],\\"component\\":[{\\"Attributes\\":{\\"amount\\":1,\\"ref\\":\\"pfx_prod_gen_e_macro\\"}}]}]}"`;

exports[`generator/helpers createWareCollection toXml() 1`] = `
"<wares>
  <ware id=\\"pfx_a\\" name=\\"a\\" transport=\\"container\\" volume=\\"1\\">
    <price min=\\"1\\" average=\\"1\\" max=\\"1\\"/>
    <production time=\\"10\\" amount=\\"1\\" method=\\"default\\" name=\\"{20206, 101}\\">
      <primary>
        <ware ware=\\"energycells\\" amount=\\"50\\"/>
      </primary>
      <effects>
        <effect type=\\"efficiency\\" product=\\"1\\"/>
      </effects>
    </production>
  </ware>
  <ware id=\\"pfx_b\\" name=\\"b\\" transport=\\"container\\" volume=\\"1\\">
    <price min=\\"1\\" average=\\"1\\" max=\\"1\\"/>
    <production time=\\"10\\" amount=\\"1\\" method=\\"default\\" name=\\"{20206, 101}\\">
      <primary>
        <ware ware=\\"energycells\\" amount=\\"50\\"/>
      </primary>
      <effects>
        <effect type=\\"efficiency\\" product=\\"1\\"/>
      </effects>
    </production>
  </ware>
  <ware id=\\"pfx_c\\" name=\\"c\\" transport=\\"container\\" volume=\\"1\\">
    <price min=\\"1\\" average=\\"1\\" max=\\"1\\"/>
    <production time=\\"10\\" amount=\\"1\\" method=\\"default\\" name=\\"{20206, 101}\\">
      <primary>
        <ware ware=\\"energycells\\" amount=\\"50\\"/>
      </primary>
      <effects>
        <effect type=\\"efficiency\\" product=\\"1\\"/>
      </effects>
    </production>
  </ware>
  <ware id=\\"pfx_d\\" name=\\"d\\" transport=\\"container\\" volume=\\"1\\">
    <price min=\\"1\\" average=\\"1\\" max=\\"1\\"/>
    <production time=\\"10\\" amount=\\"1\\" method=\\"default\\" name=\\"{20206, 101}\\">
      <primary>
        <ware ware=\\"energycells\\" amount=\\"50\\"/>
      </primary>
      <effects>
        <effect type=\\"efficiency\\" product=\\"1\\"/>
      </effects>
    </production>
  </ware>
  <ware id=\\"pfx_e\\" name=\\"e\\" transport=\\"container\\" volume=\\"1\\">
    <price min=\\"1\\" average=\\"1\\" max=\\"1\\"/>
    <production time=\\"10\\" amount=\\"1\\" method=\\"default\\" name=\\"{20206, 101}\\">
      <primary>
        <ware ware=\\"energycells\\" amount=\\"50\\"/>
      </primary>
      <effects>
        <effect type=\\"efficiency\\" product=\\"1\\"/>
      </effects>
    </production>
  </ware>
  <ware id=\\"pfx_prod_gen_a_macro_01\\" name=\\"a production\\" transport=\\"container\\" volume=\\"1\\" tags=\\"module\\" description=\\"blueprint description\\">
    <price min=\\"1\\" average=\\"1\\" max=\\"1\\"/>
    <production time=\\"791\\" amount=\\"1\\" method=\\"default\\" name=\\"{20206,101}\\">
      <primary>
        <ware ware=\\"claytronics\\" amount=\\"30\\"/>
        <ware ware=\\"energycells\\" amount=\\"75\\"/>
        <ware ware=\\"hullparts\\" amount=\\"150\\"/>
      </primary>
    </production>
    <restriction licence=\\"station_gen_basic \\"/>
    <research time=\\"10\\">
      <research>
        <ware ware=\\"research_module_production\\"/>
      </research>
    </research>
    <owner faction=\\"antigone\\"/>
    <owner faction=\\"argon\\"/>
    <owner faction=\\"holyorder\\"/>
    <owner faction=\\"paranid\\"/>
    <owner faction=\\"teladi\\"/>
    <component amount=\\"1\\" ref=\\"pfx_prod_gen_a_macro\\"/>
  </ware>
  <ware id=\\"pfx_prod_gen_b_macro_01\\" name=\\"b production\\" transport=\\"container\\" volume=\\"1\\" tags=\\"module\\" description=\\"blueprint description\\">
    <price min=\\"1\\" average=\\"1\\" max=\\"1\\"/>
    <production time=\\"791\\" amount=\\"1\\" method=\\"default\\" name=\\"{20206,101}\\">
      <primary>
        <ware ware=\\"claytronics\\" amount=\\"30\\"/>
        <ware ware=\\"energycells\\" amount=\\"75\\"/>
        <ware ware=\\"hullparts\\" amount=\\"150\\"/>
      </primary>
    </production>
    <restriction licence=\\"station_gen_basic \\"/>
    <research time=\\"10\\">
      <research>
        <ware ware=\\"research_module_production\\"/>
      </research>
    </research>
    <owner faction=\\"antigone\\"/>
    <owner faction=\\"argon\\"/>
    <owner faction=\\"holyorder\\"/>
    <owner faction=\\"paranid\\"/>
    <owner faction=\\"teladi\\"/>
    <component amount=\\"1\\" ref=\\"pfx_prod_gen_b_macro\\"/>
  </ware>
  <ware id=\\"pfx_prod_gen_c_macro_01\\" name=\\"c production\\" transport=\\"container\\" volume=\\"1\\" tags=\\"module\\" description=\\"blueprint description\\">
    <price min=\\"1\\" average=\\"1\\" max=\\"1\\"/>
    <production time=\\"791\\" amount=\\"1\\" method=\\"default\\" name=\\"{20206,101}\\">
      <primary>
        <ware ware=\\"claytronics\\" amount=\\"30\\"/>
        <ware ware=\\"energycells\\" amount=\\"75\\"/>
        <ware ware=\\"hullparts\\" amount=\\"150\\"/>
      </primary>
    </production>
    <restriction licence=\\"station_gen_basic \\"/>
    <research time=\\"10\\">
      <research>
        <ware ware=\\"research_module_production\\"/>
      </research>
    </research>
    <owner faction=\\"antigone\\"/>
    <owner faction=\\"argon\\"/>
    <owner faction=\\"holyorder\\"/>
    <owner faction=\\"paranid\\"/>
    <owner faction=\\"teladi\\"/>
    <component amount=\\"1\\" ref=\\"pfx_prod_gen_c_macro\\"/>
  </ware>
  <ware id=\\"pfx_prod_gen_d_macro_01\\" name=\\"d production\\" transport=\\"container\\" volume=\\"1\\" tags=\\"module\\" description=\\"blueprint description\\">
    <price min=\\"1\\" average=\\"1\\" max=\\"1\\"/>
    <production time=\\"791\\" amount=\\"1\\" method=\\"default\\" name=\\"{20206,101}\\">
      <primary>
        <ware ware=\\"claytronics\\" amount=\\"30\\"/>
        <ware ware=\\"energycells\\" amount=\\"75\\"/>
        <ware ware=\\"hullparts\\" amount=\\"150\\"/>
      </primary>
    </production>
    <restriction licence=\\"station_gen_basic \\"/>
    <research time=\\"10\\">
      <research>
        <ware ware=\\"research_module_production\\"/>
      </research>
    </research>
    <owner faction=\\"antigone\\"/>
    <owner faction=\\"argon\\"/>
    <owner faction=\\"holyorder\\"/>
    <owner faction=\\"paranid\\"/>
    <owner faction=\\"teladi\\"/>
    <component amount=\\"1\\" ref=\\"pfx_prod_gen_d_macro\\"/>
  </ware>
  <ware id=\\"pfx_prod_gen_e_macro_01\\" name=\\"e production\\" transport=\\"container\\" volume=\\"1\\" tags=\\"module\\" description=\\"blueprint description\\">
    <price min=\\"1\\" average=\\"1\\" max=\\"1\\"/>
    <production time=\\"791\\" amount=\\"1\\" method=\\"default\\" name=\\"{20206,101}\\">
      <primary>
        <ware ware=\\"claytronics\\" amount=\\"30\\"/>
        <ware ware=\\"energycells\\" amount=\\"75\\"/>
        <ware ware=\\"hullparts\\" amount=\\"150\\"/>
      </primary>
    </production>
    <restriction licence=\\"station_gen_basic \\"/>
    <research time=\\"10\\">
      <research>
        <ware ware=\\"research_module_production\\"/>
      </research>
    </research>
    <owner faction=\\"antigone\\"/>
    <owner faction=\\"argon\\"/>
    <owner faction=\\"holyorder\\"/>
    <owner faction=\\"paranid\\"/>
    <owner faction=\\"teladi\\"/>
    <component amount=\\"1\\" ref=\\"pfx_prod_gen_e_macro\\"/>
  </ware>
</wares>"
`;
